PSReadLine/ReadLine.cs

```diff
@@ -144,6 +144,14 @@ private void ReadOneOrMoreKeys()
                 while (_charMap.KeyAvailable)
                 {
                     var key = PSKeyInfo.FromConsoleKeyInfo(_charMap.ReadKey());
+                    // If PSReadLine is running under a host that can cancel it, the cancellation
+                    // will come at a time when ReadKey is stuck waiting for input. The next key
+                    // press will be used to force it to return, and so we want to discard this key
+                    // since we were already canceled.
+                    if (cancellationToken.IsCancellationRequested)
+                    {
+                        continue;
```

Comment:
Perhaps this should be `break`?

Reply:
> Perhaps this should be `break`?

I was debating leaving that feedback, but I think maybe it *should* drain? If the key press is translated as an escape sequence (like an arrow key with VTINPUT/non-Windows) then we don't want to abort mid-way through processing the sequence, right?

Reply:
I guess it's possible, since .NET does what it wants with the key press we send. And yes, PSES is sending `\0` but that's not a guarantee necessarily.

Reply:
I'm more imagining that the only place that `break` vs `continue` would matter here is if the pending key press was an escape sequence. In this scenario I'd guess that it's a race condition between a cancellation happening and a key press.

In that scenario if we `break` instead we'll get the remainder of the escape sequence and `\0` next key press (if I understand the code correctly)

Reply:
Yeah, I guess `break` would be fine.